Modules/mmapmodule.c

```diff
@@ -293,6 +303,69 @@ safe_memcpy(void *restrict dest, const void *restrict src, size_t count) {
 #endif
 }
 
+#if defined(MS_WIN32) && !defined(DONT_USE_SEH)
+#define handle_invalid_mem(sourcecode) \
```

Comment:
This is clever! I haven't seen it done like this before. A few suggestions:
* can we make it `HANDLE_INVALID_MEM` so it's more obviously a macro
* can we use `{` `}` around the code when it's used to make it more obviously a block
* can we put the `return 0` in the functions rather than the macro (return -1 here is fine)